### ISSUE TITLE
1.20.1 crash/datapack issue with Terralith and Cyanide 

### ISSUE THREAD

**Issue by Daltaxy** [Shadows-of-Fire/Apotheosis#1042]
Hello. I'm very sorry in advance as this will certainly be very confusing, so I'll be here to test and answer if you need anything

It all starts with 
https://github.com/Stardust-Labs-MC/Terralith/issues/49
which then redirects me to
https://github.com/Shadows-of-Fire/Apotheosis/issues/1035
where I talk about two different issues like an idiot. On one hand, datapack issue with Apotheosis when used with Cyanide and Terralith
and on the other hand, a config mismatch. 

Everything resurfaced (except the config mismatch, that's fixed) after this 
https://github.com/mekanism/Mekanism/issues/7894
which causes a crash when opening any recipe tag in JEI.

I try removing Cyanide and enabling Apotheosis (I didn't even know it was disabled), because I need to know a few recipes; it works fine until I reach a sort of modded desert biome where my game freezes. I check the log and it indicates that it crashed (Exception generating a new chunk) 
crash report for that: https://mclo.gs/fdAI1Su
I try restarting to make sure, but the game crashes on world load with the same issue.
I enable Cyanide, and now this screen shows up: 
![image](https://github.com/Stardust-Labs-MC/Terralith/assets/111076631/eafec2a2-44ca-4f86-a259-70b458afd1df)
the worst part now, is that DISABLING Apotheosis for some reason removes that datapack issue ^

In short:
Terralith + Cyanide + Apotheosis = datapack error (image above)
Terralith + Apotheosis (no Cyanide) = world crash (https://mclo.gs/AeeMjgp)
Terralith + Cyanide (no Apotheosis) = everything works fine except for the addition below
(Cyanide + Mekanism = crash when opening any recipe in jei)

So, finally, sadly, I can say there is an issue with Apotheosis or some mod that I can't see anywhere in the crash report (Pineapple Delight is already gone)

**Comment by Daltaxy** [comment]
Terralith (no Cyanide, no Apotheosis) = world crash
so Apotheosis isn't responsible for the world crash but seems to be responsible for the datapack error

**Comment by SiverDX** [comment]
feature cycle error with terralith and biomesplenty, choose one

**Comment by Daltaxy** [comment]
> feature cycle error with terralith and biomesplenty, choose one

except the terralith mod maker never pointed that out, has to be a false lead?

**Comment by SiverDX** [comment]
dont know what happened previously 
did you disable the adventure module?

```log
java.lang.IllegalStateException: Feature order cycle found, involved sources: [Reference{ResourceKey[minecraft:worldgen/biome / terralith:snowy_cherry_grove]=net.minecraft.world.level.biome.Biome@7cda6f12}, Reference{ResourceKey[minecraft:worldgen/biome / biomesoplenty:lush_savanna]=net.minecraft.world.level.biome.Biome@7f5e4ccc}]
```

**Comment by Daltaxy** [comment]
I did yeah but that was a while ago

**Comment by SiverDX** [comment]
you can only disable that if youre using neoforge
if you enable it you shouldnt run in the initial error, laving only the feature cycle one

**Comment by Daltaxy** [comment]
forge 47.2.1, isn't that neofoge?

**Comment by SiverDX** [comment]
no, forge is forge and neoforge is neoforge
if you're using forge you should use 47.1.3 as others are likely to not be compatible with neoforge mods

then again even with that version compatibility is not guaranteed

**Comment by Daltaxy** [comment]
I can't find neoforge anywhere lol, and the CarryOn dev said that I was on neoforge and that the mod won't be compatible with it

**Comment by Apollounknowndev** [comment]
A separate mod is causing the crash. I tested with Apotheosis, BOP and Terralith (and their dependencies) with no crash.

**Comment by Shadows-of-Fire** [comment]
This appears to be external and or a duplicate of #989 
NeoForge is available here https://neoforged.net/
Launchers are being a bit slow with adaptation. CF has no support, and IIRC Prism only supports it on the dev branch at the moment.
